src/formatter-settings/index.ts

```diff
@@ -1,99 +1,356 @@
+        this.profileElements.clear();
+        this.profileSettings.clear();
+        this.lastElement = undefined;
+        this.diagnosticCollection.clear();
+        const diagnostics: vscode.Diagnostic[] = [];
+        const documentDOM = new DOMParser().parseFromString(document.getText());
+        this.settingsVersion = documentDOM.documentElement.getAttribute("version") || this.settingsVersion;
+        const profiles = documentDOM.documentElement.getElementsByTagName("profile");
+        if (!profiles || profiles.length === 0) {
+            diagnostics.push(new vscode.Diagnostic(new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 0)), "No valid profiles found."));
+            return diagnostics;
+        }
+        const settingsProfileName: string | undefined = vscode.workspace.getConfiguration("java").get<string>(JavaConstants.SETTINGS_PROFILE_KEY);
+        for (let i = 0; i < profiles.length; i++) {
+            if (!settingsProfileName || settingsProfileName === profiles[i].getAttribute("name")) {
+                if (profiles[i].getAttribute("kind") !== "CodeFormatterProfile") {
+                    continue;
+                }
+                this.settingsVersion = profiles[i].getAttribute("version") || this.settingsVersion;
+                const settings = profiles[i].getElementsByTagName("setting");
+                for (let j = 0; j < settings.length; j++) {
+                    const setting: DOMElement = settings[j] as DOMElement;
+                    const settingContent: string = new XMLSerializer().serializeToString(setting);
+                    const id = setting.getAttribute("id");
+                    if (!id) {
+                        diagnostics.push(new vscode.Diagnostic(new vscode.Range(new vscode.Position(setting.lineNumber - 1, setting.columnNumber - 1), new vscode.Position(setting.lineNumber - 1, setting.columnNumber - 1 + settingContent.length)), "The setting has no 'id' property.", vscode.DiagnosticSeverity.Error));
+                        continue;
+                    }
+                    const value = settings[j].getAttribute("value");
+                    if (!value) {
+                        diagnostics.push(new vscode.Diagnostic(new vscode.Range(new vscode.Position(setting.lineNumber - 1, setting.columnNumber - 1), new vscode.Position(setting.lineNumber - 1, setting.columnNumber - 1 + settingContent.length)), "The setting has no 'value' property.", vscode.DiagnosticSeverity.Error));
+                        continue;
+                    }
+                    this.profileElements.set(id, setting);
+                    this.profileSettings.set(id, value);
+                    this.lastElement = setting;
+                }
+                break;
+            }
+        }
+        if (!this.profileElements.size) {
+            diagnostics.push(new vscode.Diagnostic(new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 0)), "No valid settings found in the profile."));
```

Comment:
what's the default DiagnosticSeverity

Reply:
The default severity is `Error`. Here we just push an `Error` since it's meaningless to use an empty profile.